src/codegen/compile_enhanced.ml

```diff
@@ -12320,6 +12326,13 @@ and compile_prim_invocation (env : E.t) ae p es at =
   | SystemCyclesBurnPrim, [e1] ->
     SR.Vanilla, compile_exp_vanilla env ae e1 ^^ Cycles.burn env
 
+  | SystemTimeoutSetPrim, [e1] ->
+    SR.unit,
+    compile_exp_as env ae (SR.UnboxedWord64 Type.Nat32) e1 ^^
+    TaggedSmallWord.lsb_adjust Type.Nat32 ^^
```

Comment:
```suggestion
    TaggedSmallWord.lsb_adjust Type.Nat32 ^^
```
This looks dubious to me.  Will need to review carefully.

Reply:
Look at `OtherPrim "setCandidLimits"` for a similar example. If the significant bits were in the lower portion, this would always send `0` to the RTS call. @luc-blaeser can you have a look too?

Reply:
Ok, you are right it is fine.
I was confused because StackRep.adjust calls TaggedSmallWord.untag, but that only clears the tag bits, it doesn't shift down. Unlike BitTagged.untag, which shifts....